compiler/rustc_hir_analysis/src/collect/item_bounds.rs

```diff
@@ -45,6 +45,13 @@ fn associated_type_bounds<'tcx>(
             | PredicateFilter::SelfTraitThatDefines(_)
             | PredicateFilter::SelfAndAssociatedTypeBounds => {
                 icx.lowerer().add_default_traits(&mut bounds, item_ty, hir_bounds, None, span);
+                icx.lowerer().adjust_sizedness_params_and_assoc_types(
```

Comment:
All the other sites adjust sizedness before adding default bounds and right after lower_bounds. May be worth a comment here

Reply:
it is kind of annoying that every `lower_bounds` call has to be paired with an adjust call, but with subtle differences

Reply:
I've reordered these, the difference wasn't significant. I've also pulled forward a change from Part II that removes `adjust_sizedness_predicate` (where I removed `PointeeSized`) and just puts that in `lower_bounds` to avoid lowering any of those predicates in the first place (in Part II, I also need to migrate `Sized` to `const Sized`, which ended up being another change that made sense in `lower_bounds` to make sure it always happened).